File: cvgen-api/src/main/java/dev/carlosmz96/cvgen/cvgen_api/models/entities/Curriculum.java
package dev.carlosmz96.cvgen.cvgen_api.models.entities;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import dev.carlosmz96.cvgen.cvgen_api.models.entities.security.User;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "curriculum")
public class Curriculum {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String summary;
    private String theme;
    private String language;
    private LocalDate createdAt;
    private LocalDate updatedAt;

    @ManyToOne
    private User user;

    @OneToMany(mappedBy = "curriculum", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Experience> experiences = new ArrayList<>();

    @OneToMany(mappedBy = "curriculum", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Education> educations = new ArrayList<>();

    @OneToMany(mappedBy = "curriculum", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Skill> skills = new ArrayList<>();

    @OneToMany(mappedBy = "curriculum", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Certification> certifications = new ArrayList<>();

    @OneToMany(mappedBy = "curriculum", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<LanguageSkill> languageSkills = new ArrayList<>();

}

File: cvgen-client/src/app/models/Curriculum.ts
import { Certification } from "./Certification";
import { Education } from "./Education";
import { Experience } from "./Experience";
import { LanguageSkill } from "./LanguageSkill";
import { Skill } from "./Skill";
import { User } from "./User";

export interface Curriculum {
  id?: number;
  country: string;
  city: string;
  phone?: string;
  title: string;
  summary?: string;
  theme: string;
  language: string;
  user?: User;
  experiences?: Experience[];
  educations?: Education[];
  skills?: Skill[];
  certifications?: Certification[];
  languageSkills?: LanguageSkill[];
}
